riposte-core/src/main/java/com/nike/riposte/server/channelpipeline/HttpChannelInitializer.java

```diff
@@ -430,9 +451,9 @@ public void initChannel(SocketChannel ch) {
                       new OpenChannelLimitHandler(openChannelsGroup, maxOpenChannelsThreshold));
         }
 
-        // INBOUND - Add the RequestFilterHandler (if we have any filters to apply).
-        if (cachedRequestFilterHandler != null)
-            p.addLast(REQUEST_FILTER_HANDLER_NAME, cachedRequestFilterHandler);
+        // INBOUND - Add the RequestFilterHandler for before security (if we have any filters to apply).
+        if (beforeSecurityRequestFilterHandler != null)
+            p.addLast(REQUEST_FILTER_HANDLER_NAME, beforeSecurityRequestFilterHandler);
```

Comment:
Great catch! I have renamed those as suggested. I was not receiving the IllegalArgumentException at that time due to mocking.

As discussed, I have added a component test to verify the actual behavior (this would have thrown the IllegalArgumentException).